### ISSUE TITLE
[Bug Report] ttnn.transpose forcibly changes dimension size with tile-aligned size

### ISSUE THREAD

**Issue by marty1885** [tenstorrent/tt-metal#11650]
**Describe the bug**
When transposing a tensor of shape `[1, 5, 10[32], 15[32]]` by swapping axis 1 and 2. TTNN uses the tile-aligned shape instead of the real underlying shape. This bug is discovered by GGML's unit tests.

This is a minimal reproducible example

```python
import ttnn
from tt_lib import tensor
import torch
device = ttnn.open_device(0)

shape = (1, 5, 10, 15)
tensor = torch.rand(*shape, dtype=torch.float32)
a = ttnn.from_torch(tensor, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
b = ttnn.transpose(a, 1, 2)

print("Torch shape", torch.transpose(tensor, 1, 2).shape)
print("TTNN shape", b.shape)
```

output

```
                 Device | INFO     | Opening user mode device driver
2024-08-20 05:19:44.264 | INFO     | SiliconDriver   - Detected 1 PCI device : [0]
                  Metal | INFO     | Initializing device 0. Program cache is NOT enabled
                  Metal | INFO     | AI CLK for device 0 is:   1202 MHz
Torch shape torch.Size([1, 10, 5, 15])
TTNN shape ttnn.Shape([1, 32, 5, 32])
                  Metal | INFO     | Closing device 0
                  Metal | INFO     | Disabling and clearing program cache on device 0
                 Device | INFO     | Closing user mode device drivers
```

**To Reproduce**
1. Run the above program
2. See the output

**Expected behavior**
ttnn.transpose should output tensor with\ correct shape

**Screenshots**

**Please complete the following environment information:**
- OS: Ubuntu 22.04
- Commit: 046237fd9c24f51fefd05f66c270c78e606eae85

**Additional context**

**Comment by davorchap** [comment]
we're on it

**Comment by marty1885** [comment]
@davorchap Nice to hear that!

Regarding the tile size being implementation detail - does that mean TTNN has to be compiled to have a different tile size or is it a detail to the kernels?

**Comment by jvasilje** [comment]
Added the P0 label on this. 

This should get fixed with two work streams that are in progress:
- a new and more flexible tensor shape in TT-NN 
- generality work in transpose 

@ntarafdar please post updates here.

**Comment by ntarafdar** [comment]
This might have been covered with the generality push in transpose. Will get more info tomorrow
